gems/aws-sdk-s3/lib/aws-sdk-s3/presigned_post.rb

```diff
@@ -582,15 +611,16 @@ def check_required_values!
       def bucket_url
         url = Aws::Partitions::EndpointProvider.resolve(@bucket_region, 's3')
         url = URI.parse(url)
-        if Plugins::BucketDns.dns_compatible?(@bucket_name, true)
-          url.host = @bucket_name + '.' + url.host
+        if Plugins::BucketDns.dns_compatible?(@bucket_name, _ssl = true)
+          url.host = "#{@bucket_name}.#{url.host}"
         else
-          url.path = '/' + @bucket_name
+          url.path = "/#{@bucket_name}"
         end
         if @bucket_region == 'us-east-1'
           # keep legacy behavior by default
           url.host = Plugins::IADRegionalEndpoint.legacy_host(url.host)
         end
+        url.host = "#{@bucket_name}.s3-accelerate.amazonaws.com" if @accelerate
```

Comment:
Does this ever conflict with the legacy behavior above for us-east-1?

Reply:
It shouldn't. s3 accelerate isn't regional. The legacy behavior will override the host so I have to do this after that. `legacy_host` basically does a gsub of `us-east-1` out of the host.